test/extended/etcd/helpers/helpers.go

```diff
@@ -261,13 +253,61 @@ func EnsureReadyReplicasOnCPMS(ctx context.Context, t TestingT, expectedReplicaC
 			t.Logf("expected %d ready replicas on CPMS, got: %v,", expectedReplicaCount, cpms.Status.ReadyReplicas)
 			return false, nil
 		}
-
 		t.Logf("CPMS has reached the desired number of ready replicas: %v,", cpms.Status.ReadyReplicas)
 
+		err = EnsureReadyMasterNodes(ctx, expectedReplicaCount, nodeClient)
+		if err != nil {
+			t.Logf("expected number of master nodes is not ready yet: '%w'", err)
+			return false, nil
+		}
+
+		err = EnsureCPMSReplicasConverged(ctx, cpmsClient)
+		if err != nil {
+			t.Logf("%w", err)
+			return false, nil
+		}
```

Comment:
This is the wrong place for this check. What I meant by adding a check for `EnsureCPMSReplicasConverged` was as the last step in the test. Because during scale-up we actually expect `Spec.Replicas=3` and `Spec.ReadyReplicas=4` so they shouldn't be equal.

Move this check to L112 in [test/extended/etcd/vertical_scaling.go](https://github.com/openshift/origin/pull/27788/files#diff-f4228c2c6437654a567b721f59725242c4209fefa4515c65ec49c61959e77da8)